x/blockdb/database_test.go

```diff
@@ -0,0 +1,420 @@
+			indexPath := filepath.Join(tt.config.IndexDir, indexFileName)
+			require.FileExists(t, indexPath)
+
+			// Test that we can close the database
+			require.NoError(t, db.Close())
+		})
+	}
+}
+
+func TestNew_IndexFileErrors(t *testing.T) {
+	tests := []struct {
+		name       string
+		setup      func() (string, string)
+		wantErrMsg string
+	}{
+		{
+			name: "corrupted index file",
+			setup: func() (string, string) {
+				tempDir := t.TempDir()
+				indexDir := filepath.Join(tempDir, "index")
+				dataDir := filepath.Join(tempDir, "data")
+				require.NoError(t, os.MkdirAll(indexDir, 0o755))
+				require.NoError(t, os.MkdirAll(dataDir, 0o755))
+
+				// Create a corrupted index file
+				indexPath := filepath.Join(indexDir, indexFileName)
+				corruptedData := []byte("corrupted index file data")
+				require.NoError(t, os.WriteFile(indexPath, corruptedData, defaultFilePermissions))
+
+				return indexDir, dataDir
+			},
+			wantErrMsg: "failed to read index header",
+		},
+		{
+			name: "version mismatch in existing index file",
+			setup: func() (string, string) {
+				tempDir := t.TempDir()
+				indexDir := filepath.Join(tempDir, "index")
+				dataDir := filepath.Join(tempDir, "data")
+
+				// Create directories
+				require.NoError(t, os.MkdirAll(indexDir, 0o755))
+				require.NoError(t, os.MkdirAll(dataDir, 0o755))
+
+				// Create a valid index file with wrong version
+				indexPath := filepath.Join(indexDir, indexFileName)
+				header := indexFileHeader{
+					Version:             999, // Wrong version
+					MinHeight:           0,
+					MaxDataFileSize:     DefaultMaxDataFileSize,
+					MaxHeight:           unsetHeight,
+					MaxContiguousHeight: unsetHeight,
+					NextWriteOffset:     0,
+				}
+
+				headerBytes, err := header.MarshalBinary()
+				require.NoError(t, err)
+				require.NoError(t, os.WriteFile(indexPath, headerBytes, defaultFilePermissions))
+
+				return indexDir, dataDir
+			},
+			wantErrMsg: "mismatched index file version",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			indexDir, dataDir := tt.setup()
+			if indexDir == "" || dataDir == "" {
+				t.Skip("Setup failed, skipping test")
+			}
+
+			config := DefaultConfig().WithIndexDir(indexDir).WithDataDir(dataDir)
+			_, err := New(config, logging.NoLog{})
+			require.Contains(t, err.Error(), tt.wantErrMsg)
+		})
+	}
+}
+
+func TestIndexFileHeaderAlignment(t *testing.T) {
+	require.Equal(t, uint64(0), sizeOfIndexFileHeader%sizeOfIndexEntry,
+		"sizeOfIndexFileHeader (%d) is not a multiple of sizeOfIndexEntry (%d)",
+		sizeOfIndexFileHeader, sizeOfIndexEntry)
+}
+
+func TestIndexEntrySizePowerOfTwo(t *testing.T) {
+	// Check that sizeOfIndexEntry is a power of 2
+	// This is important for memory alignment and performance
+	require.Equal(t, uint64(0), sizeOfIndexEntry&(sizeOfIndexEntry-1),
+		"sizeOfIndexEntry (%d) is not a power of 2", sizeOfIndexEntry)
+}
+
+func TestNew_IndexFileConfigPrecedence(t *testing.T) {
```

Comment:
This test raises a good point - shouldn't we log an info in case the file config differs from the software config? Otherwise we will get weird behavior: We first log the software config and then proceed to overwrite it silently.